File: oss/src/main/java/io/tn/oss/amazon/S3PrincipalArgs.java
package io.tn.oss.amazon;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class S3PrincipalArgs {
    @JsonProperty("AWS")
    @SerializedName("AWS")
    private List<String> aws = new ArrayList<>();
}

File: oss/src/main/kotlin/net/yan100/compose/oss/model/S3PrincipalArgs.kt
package net.yan100.compose.oss.model

import com.fasterxml.jackson.annotation.JsonProperty
import com.google.gson.annotations.SerializedName

data class S3PrincipalArgs(
  @JsonProperty("AWS")
  @SerializedName("AWS")
  var aws: MutableList<String> = ArrayList()
)
